### DISCUSSION TITLE
rsync errors

### DISCUSSION THREAD

**Discussion by meganewing** [sr320/course-fish546-2021#80] [Q&A]
Why are some reasons that rsync might run into errors? 

Only a few of my samples (out of 23 total) returned unchanged md5sums after rsync-ing. The rest were changed and some even had changed file names. Some refused to copy at all. 

Rsync code (from terminal): https://github.com/fish546-2021/Megan-project/blob/main/code/0208-rsyncFilesToExternalStorage-terminal.txt

md5sum code: https://github.com/fish546-2021/Megan-project/blob/main/code/0209-md5sums.ipynb

md5sum output: https://github.com/fish546-2021/Megan-project/blob/main/analyses/0209-md5sums.txt
what md5sums should be: https://github.com/fish546-2021/Megan-project/blob/main/analyses/0128-Giles-md5sums.txt

**Comment by kubu4** [comment]
Before I address the error message that you're getting can you please describe what you're trying to do? It helps for you to verbalize it and it will help us get a better understanding of what you're trying to accomplish.

Anyway, here's your command:

`rsync -avzh fish546@ostrich.fish.washington.edu:/Users/fish546/Documents/Files_Giles_Uploaded_For_Chris /Volumes/Samsung\ USB/salmon/`

This command is trying to sync files from Ostrich to a USB drive on your computer.

The error you're getting that points to the actual problem with the transfer is this:

`rsync: write failed on "/Volumes/Samsung USB/salmon/Files_Giles_Uploaded_For_Chris/17104-02RT-01-14_S22_L002_R2_001.fastq.gz": File too large (27)`

The key point of this error is this part: `File too large (27)`

Your USB drive is out of storage space, so you're unable to copy more files.

Dont' worry about the MD5 checkums for now. Glad you checked them, though!

**Comment by kubu4** [comment]
Actually, the reason might be related to the format of your external USB drive, as opposed to available storage space... There are formatting options (e.g. FAT32) that do not allow files large than 4GB. Might also have to take that into consideration to resolve this.

What is the main reason you're trying to transfer these files to your personal computer? If it's for Jupyter Notebook usage, we can discuss options on how to do this on a remote computer (i.e. Ostrich) so that you don't have to rely on your personal computer.
